app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -123,6 +125,17 @@ protected void onCreate(Bundle savedInstanceState) {
             }
         }
         WebViewProvider.preload(this);
+
+        if(sIsNewCreated && (!Settings.getInstance(this).didShowRateAppDialog() || !Settings.getInstance(this).didShowShareAppDialog())) {
+            sIsNewCreated = false;
+            Settings.getInstance(this).increaseAppCreateCounter();
+            if(!Settings.getInstance(this).didShowRateAppDialog() && Settings.getInstance(this).getAppCreateCount() == DialogUtils.APP_CREATE_THRESHOLD_FOR_RATE_APP) {
```

Comment:
Maybe we can go with >= DialogUtils.APP_CREATE_THRESHOLD_FOR_RATE_APP to make sure the dialog will eventually be shown at least once even when some exceptions are thrown before we set Settings.getInstance(this).didShowRateAppDialog() to true.

Reply:
Sounds reasonable. I'll have a patch for this.